c-dependencies/js-compute-runtime/js-compute-builtins.cpp

```diff
@@ -3209,28 +3222,33 @@ namespace FetchEvent {
 
     RootedObject promise(cx, JS::ReadableStreamDefaultReaderRead(cx, reader));
     if (!promise) return false;
-    if (!JS::AddPromiseReactionsIgnoringUnhandledRejection(cx, promise,
-                                                           then_handler, catch_handler))
-    {
+    if (!JS::AddPromiseReactions(cx, promise, then_handler, catch_handler))
       return false;
-    }
 
     if (!start_response(cx, response_obj, true))
       return false;
 
     return true;
   }
 
-  bool respondWith(JSContext* cx, unsigned argc, Value* vp) {
-    METHOD_HEADER(1)
+  // Steps in this function refer to the spec at
+  // https://w3c.github.io/ServiceWorker/#fetch-event-respondwith
+  bool then_handler(JSContext* cx, unsigned argc, Value* vp) {
+    CallArgs args = CallArgsFromVp(argc, vp);
+    RootedObject event(cx, &js::GetFunctionNativeReserved(&args.callee(), 0).toObject());
 
-    // TODO: support Promise<Response>
+    // Step 10.1
     if (!Response::is_instance(args.get(0))) {
```

Comment:
This was a bit confusing at first given the error message just below referring to promises. Could you add a comment here pointing to `respondWith` below and indicating that any promises are resolved by this point?